app/ui/src/app/integrations/edit-page/action-select/action-select.component.ts

```diff
@@ -101,16 +102,48 @@ export class IntegrationsSelectActionComponent extends FlowPage
     }
   }
 
+  // filters out actions that do not use the 'From' pattern
+  connectorStartActionsFilter(connector: Connector): Connector {
+    const filteredConnectorActions = connector.actions.filter(action => action.pattern === 'From');
+    connector.actions = filteredConnectorActions;
+    return connector;
+  }
+
+  // filters out actions that do not use the 'To' pattern
+  connectorFinishActionsFilter(connector: Connector): Connector {
+    const filteredConnectorActions = connector.actions.filter(action => action.pattern === 'To');
+    connector.actions = filteredConnectorActions;
+    return connector;
+  }
+
   ngOnInit() {
-    this.actions = this.connector
-      .filter(connector => connector !== undefined)
-      .map(connector => connector.actions);
+    // set a local var for current step
+    this.route.params.pluck<Params, string>('position').subscribe(value => this.currentStep = parseInt(value));
```

Comment:
This is not the Angular way to fetch params from URLs. In first place please remember that route information (URL path fragments, querystring params) is meant to be aynchronous. Everything's a stream in Angular, including routes.

If you feel that users will always load this page or whatever it is from another unrelated component/page, then perhaps you can get away with inspecting the route snapshot in a synchronous fashion, like this:

```typescript
this.currentStep = +this.route.snapshot.paramMap.get('position')
```

This is important since the interface type of the `params` property can change anytime from a StringMap to whatever the Angular team comes up with and your code will break. The `+` will turn the returned string into the number you need. And please, never use `parseInt(..., 10)` without the base index (10 for integers). Otherwise you might confront unexpected results depending on the structure of the parsed string.

If you prefer to stay async for whatever reason, then try this:

```typescript
this.route.paramMap.subscribe(params => this.currentStep = +params.get('position'))
```